redaxo/include/classes/class.rex_sql.inc.php

```diff
@@ -558,7 +558,8 @@ class rex_sql
      */
     /*public*/ function hasNext()
     {
-        return $this->counter != $this->rows;
+        #return $this->counter != $this->rows;
+        return (($this->counter) < ($this->rows-1));
```

Comment:
Am 04.06.2015 um 19:33 schrieb Markus Staab:
>
> In redaxo/include/classes/class.rex_sql.inc.php 
> <https://github.com/redaxo/redaxo4/pull/392#discussion_r31743972>:
>
> > @@ -558,7 +558,8 @@ class rex_sql
> >       */
> >      /*public*/ function hasNext()
> >      {
> > -        return $this->counter != $this->rows;
> > +        #return $this->counter != $this->rows;
> > +        return (($this->counter) < ($this->rows-1));
>
> Shouldnt it be either |counter < rows| or |counter <= rows-1|? Counter 
> is zero based and rows is 1 based..
>
 >> Example:
- I have 6 rows and the counter is set to the last record (so counter is 5)
- I ask $sql->hasNext() and get true, because 5 < 6. But hasNext should 
return false.

> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/redaxo/redaxo4/pull/392/files#r31743972>.
>


-- 
_______________________________________

Stefan Dietl
SD-Soft - Software und Webdesign
Ortweinstraße 41
I-39024 Mals
Tel. +39 0473 835299

www.sdsoft.it
_______________________________________